transactionoutbox-core/src/main/java/com/gruelbox/transactionoutbox/DefaultInvocationSerializer.java

```diff
@@ -329,6 +338,112 @@ private String nameForClass(Class<?> clazz) {
     }
   }
 
+  static final class LocalDateTimeTypeAdapter extends TypeAdapter<LocalDateTime> {
+
+    @Override
+    public void write(JsonWriter out, LocalDateTime value) throws IOException {
+      out.value(value.format(DateTimeFormatter.ISO_LOCAL_DATE_TIME));
+    }
+
+    @Override
+    public LocalDateTime read(JsonReader in) throws IOException {
+      return LocalDateTime.parse(in.nextString(), DateTimeFormatter.ISO_LOCAL_DATE_TIME);
+    }
+  }
+
+  static final class InstantTypeAdapter extends TypeAdapter<Instant> {
+
+    @Override
+    public void write(JsonWriter out, Instant value) throws IOException {
+      out.value(DateTimeFormatter.ISO_INSTANT.format(value));
+    }
+
+    @Override
+    public Instant read(JsonReader in) throws IOException {
+      return DateTimeFormatter.ISO_INSTANT.parse(in.nextString(), Instant::from);
+    }
+  }
+
+  static final class DurationTypeAdapter extends TypeAdapter<Duration> {
+
+    @Override
+    public void write(JsonWriter out, Duration value) throws IOException {
+      out.value(value.get(ChronoUnit.SECONDS));
+    }
+
+    @Override
+    public Duration read(JsonReader in) throws IOException {
+      return Duration.of(in.nextLong(), ChronoUnit.SECONDS);
+    }
+  }
+
+  static final class LocalDateTypeAdapter extends TypeAdapter<LocalDate> {
+
+    @Override
+    public void write(JsonWriter out, LocalDate value) throws IOException {
+      out.value(DateTimeFormatter.ISO_LOCAL_DATE.format(value));
+    }
+
+    @Override
+    public LocalDate read(JsonReader in) throws IOException {
+      return DateTimeFormatter.ISO_LOCAL_DATE.parse(in.nextString(), LocalDate::from);
+    }
+  }
+
+  static final class MonthDayTypeAdapter extends TypeAdapter<MonthDay> {
+
+    private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d/M");
+
+    @Override
+    public void write(JsonWriter out, MonthDay value) throws IOException {
+      out.value(value.format(formatter));
+    }
+
+    @Override
+    public MonthDay read(JsonReader in) throws IOException {
+      return MonthDay.parse(in.nextString(), formatter);
+    }
+  }
+
+  static final class PeriodTypeAdapter extends TypeAdapter<Period> {
```

Comment:
A few of these are basically identical and could just be a `ToStringTypeAdapter`

Reply:
but `read` method  won't be easy to implement